addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -82,17 +82,24 @@ function _buildTitleElement(title) {
  * @param {string} src
  * @returns {HTMLElement}
  */
+const _buildImgElementCache = {};
 async function _buildImgElement(src) {
+    if (src in _buildImgElementCache)
+        return _buildImgElementCache[src];
+    let result;
     if (src.split('.').pop() === 'svg') {
         const response = await window.fetch(src);
         const text = await response.text();
         const parser = new window.DOMParser();
         const xmlDoc = parser.parseFromString(text, 'text/xml');
-        return xmlDoc.getElementsByTagName('svg')[0];
+        result = xmlDoc.getElementsByTagName('svg')[0];
+    } else {
+        const imgEl = document.createElement('img');
+        imgEl.src = src;
+        result = imgEl;
```

Comment:
In fact, I don't think your cache is very meaningful, especially for img tags (since you will have to clone them anyway thus requesting the src again)... the browser cache should just be enough if the Odoo server is configured properly to cache static content (and the browser would thus cache those svg and images between editing sessions too). I guess this does not hurt for local databases though...

Reply:
The problem of browser cache is that it's used only after getting 304 response from the server, which could take time

Reply:
If I am not mistaken, if there is a cache, there is no call to the server at all. The 304 is only sent if the cached resource is stale.

Reply:
Well, if most of the deployment (including runbot :wink: ) don't have cache configuration, I'd prefer to adjust code to avoid buggy/slow UI

Reply:
Most of the deployment should as per the install guidelines :wink: And runbot does as well, see this header for svg icons loading: `Cache-Control | max-age=604800, public`... it does not seem to work though, it may be worth investigating. I don't know the fetch API very well, maybe there is something to do to make it work?

Reply:
In any case, it's still ok to have a local cache for that for local instances :+1:

Reply:
I tried again on runbot cache seems work with images 

![image](https://user-images.githubusercontent.com/186131/99695904-67323b80-2a8e-11eb-868e-353f791485ae.png)

So, it's only ``get_authorized_fields`` may need to be cached... But I noticed that in  ``get_authorized_fields`` is not a problem in master branch. I'll check whether we can back port those changes to v14

Reply:
> But I noticed that in get_authorized_fields is not a problem in master branch. 

It's wrong statement. The behaviour is the same in 14 and master: it's loaded on first click on the field in form, but for every field